src/app/qgsvectorlayerdigitizingproperties.cpp

```diff
@@ -43,9 +43,8 @@ QgsVectorLayerDigitizingPropertiesPage::QgsVectorLayerDigitizingPropertiesPage(
     mGeometryPrecisionLineEdit->setValidator( new QDoubleValidator( mGeometryPrecisionLineEdit ) );
 
     const double precision( vlayer->geometryOptions()->geometryPrecision() );
-    const bool ok = true;
-    QString precisionStr( QLocale().toString( precision, ok ) );
-    if ( precision == 0.0 || ! ok )
+    QString precisionStr( QLocale().toString( precision, 'g', 15 ) );
```

Comment:
```suggestion
    QString precisionStr( QLocale().toString( precision, 'g', 17 ) );
```

Please 17

As I've written elsewhere, it doesn't make sense to have so many numbers, but that's what's used for the WKT export.


This could be the subject of an option (double even, one for Cartesian (3-4 for example) and geo (6-8 ) https://xkcd.com/2170/) to avoid having a number that drags hard in the code.

Reply:
I agree, but since you are already storing the full number with all its decimals, I don't see why you shouldn't have the option to view it fully.